### PR TITLE
pubsub_subscription RDBMS + tests

### PR DESCRIPTION
**Pull request by fen-pl**
This PR combines #2157 and #2161 as they are closely related. It also includes a commit with query optimisation.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[6021.1](https://travis-ci.org/esl/MongooseIM/jobs/467480733) / Erlang 19.3 / small_tests / 7eb9a8f82423f473087e1387239814fa0a7b8b16
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2165/6021/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2165/6021/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-12-13_12.45.22/lib.mongooseim.logs/run.2018-12-13_12.45.23/suite.log.html)

**Comment by mongoose-im** [comment]
[6023.1](https://travis-ci.org/esl/MongooseIM/jobs/467876036) / Erlang 19.3 / small_tests / 6493fa707a7877bf3c8a46d3ed4c3ba224fda9fb
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2165/6023/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2165/6023/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-12-14_07.26.15/lib.mongooseim.logs/run.2018-12-14_07.26.16/suite.log.html)

File: src/pubsub/mod_pubsub_db_rdbms.erl [hunk 1]
```diff
@@ -432,18 +432,24 @@ get_affiliation(Nidx, { LU, LS, _ }) ->
...
     ok.
 
 -spec set_subscription_opts(Nidx :: mod_pubsub:nodeIdx(),
                             LJID :: jid:ljid(),
                             SubId :: mod_pubsub:subId(),
-                            Opts :: mod_pubsub:subOptions()) -> ok.
-set_subscription_opts(Nidx, LJID, SubId, Opts) ->
-    mod_pubsub_db_mnesia:set_subscription_opts(Nidx, LJID, SubId, Opts).
+                            Opts :: mod_pubsub:subOptions()) -> {result, []}.
+set_subscription_opts(Nidx, { LU, LS, LR }, SubId, Opts) ->
```
Comment (michalwski) [inline src/pubsub/mod_pubsub_db_rdbms.erl]: According to the spec, the function returns `{result, []}` while in the code I can see it returns just `ok`.

**Comment by mongoose-im** [comment]
[6025.1](https://travis-ci.org/esl/MongooseIM/jobs/467888298) / Erlang 19.3 / small_tests / 49233116d7da6bb9593bf37eca61b82f57adc058
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2165/6025/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2165/6025/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-12-14_08.44.04/lib.mongooseim.logs/run.2018-12-14_08.44.05/suite.log.html)

**Pull request merged**